robot/Cumulus/resources/ContactPageObject.py

```diff
@@ -26,14 +25,41 @@ def _is_current_page(self):
         self.selenium.wait_until_location_contains("/view", timeout=60, message="Detail page did not load in 1 min")
         self.selenium.location_should_contain("/lightning/r/Contact/",message="Current page is not a Contact record detail view")
         
-
     def update_field_value(self,field_name,old_value,new_value):
         """Delete the old value in specified field by clicking on delete icon and update with new value"""
         locator=npsp_lex_locators['delete_icon'].format(field_name,old_value)
         self.selenium.get_webelement(locator).click() 
         self.salesforce.populate_lookup_field(field_name,new_value)
-        
-            
+
+    def go_to_related_engagement_actionplans_page(self,customerid):
+
+        """ Navigates to the related engagement plans listing page"""
+        url = self.cumulusci.org.lightning_base_url
+        name = "Action_Plans__r"
+        object_name = "{}{}".format(self.cumulusci.get_namespace_prefix(), name)
```

Comment:
I see the utilization of url,object_name & name being used in more than one place, you can create a keyword that takes in these values? or is there a specific reason behind mentioning these values in multiple places?

Reply:
Not sure because, in each of the scenarios, I am generating a new custom url and also waiting on a relevant item to load.

Reply:
I see the base url value to be `self.cumulusci.org.lightning_base_url` in all the three files which have commonality. So probably create a keyword like get_url_object_name and pass in name value and then return 2 values (base url & object_name value) and in turn, use it in formatting the custom url? You know the functionality better but by the look of it I see a pattern so I am suggesting the above. I know that is a lot of text, in short this is what I meant,
def get_url_object_name(self,name):
        base_url = self.cumulusci.org.lightning_base_url
        object_name = "{}{}".format(self.cumulusci.get_namespace_prefix(), name)
        return base_url,object_name

Reply:
thanks, added a keyword to handle the same and updates the relevant references